### DISCUSSION TITLE
Lost all my commits

### DISCUSSION THREAD

**Discussion by ressuman** [community/community#143388] [Repositories]
### Select Topic Area

General

### Body

I had to push my newly created folder(react)-client, though i had several commits of backend codes-server. In doing so i couldn't push so i used the force command and lost all commits and history of my server codes. Please i need assistance to help me recover all my commits lost. Thank you.

**Comment by abdelkarimain** [comment]
Git usually keeps a record of these commits locally even after a force push. 
1- First, see if we can find your lost commits using 
```bash 
git reflog
```
2- Look through the reflog output for the last known good state of your server code.
Once you identify the correct commit, you can recover it using
```bash
git reset --hard [commit-id]
```
hope that helps

**Comment by AyhemKouki** [comment]
You may be able to recover your lost commits by following these steps:

Check Local Reflog: Run git reflog in your repository. The reflog records all actions on your local branches, including commits, even after a forced push. Look for commit hashes before the force push.

Restore from a Commit Hash: Once you find the commit hash for a previous version of your server code:

Run git checkout <commit-hash> to check out that commit.
Create a new branch from it by running git branch recovery-branch.
Switch to the branch with git checkout recovery-branch.
Push the Recovered Branch: Once you confirm the recovered state, push it to GitHub:
git push origin recovery-branch
This branch will have all your previous commits, and you can merge it back if needed.
